waltid-services/waltid-verifier-api/src/main/kotlin/id/walt/verifier/VerifierApi.kt

```diff
@@ -256,46 +272,54 @@ fun Application.verfierApi() {
+                            logger.debug(it) { "Verification failed ($it)" }
+                            var errorDescription = it.localizedMessage
+                            logger.error { "Error: $errorDescription" }
+                            if (sessionId != null) {
+                                val session = verificationUseCase.getSession(sessionId)
+                                if (session.walletInitiatedAuthState != null) {
+                                    val state = session.walletInitiatedAuthState
+                                    when (it.localizedMessage) {
+                                        "Verification policies did not succeed: expired" -> errorDescription =
+                                            "<\$presentation_submission.descriptor_map[x].id> is expired"
 
-                                    "Verification policies did not succeed: not-before" -> errorDescription =
-                                        "<\$presentation_submission.descriptor_map[x].id> is not yet valid"
+                                        "Verification policies did not succeed: not-before" -> errorDescription =
+                                            "<\$presentation_submission.descriptor_map[x].id> is not yet valid"
 
-                                    "Verification policies did not succeed: revoked_status_list" -> errorDescription =
-                                        "<\$presentation_submission.descriptor_map[x].id> is revoked"
+                                        "Verification policies did not succeed: revoked_status_list" -> errorDescription =
+                                            "<\$presentation_submission.descriptor_map[x].id> is revoked"
+                                    }
+                                    context.respondRedirect("openid://?state=$state&error=invalid_request&error_description=$errorDescription")
                                 }
-                                context.respondRedirect("openid://?state=$state&error=invalid_request&error_description=$errorDescription")
+                            } else {
+                                logger.error(it) { "/verify error: $errorDescription" }
+                                call.respond(HttpStatusCode.BadRequest, errorDescription)
                             }
-                        } else {
-                            logger.error(it) { "/verify error: $errorDescription" }
-                            call.respond(HttpStatusCode.BadRequest, errorDescription)
+                        }.also {
+                            sessionId?.run { verificationUseCase.notifySubscribers(this) }
                         }
-                    }.also {
-                        sessionId?.run { verificationUseCase.notifySubscribers(this) }
+                }.onFailure { e ->
+                    when (e) {
+                        is IllegalArgumentException -> call.respond(HttpStatusCode.BadRequest, e.localizedMessage)
```

Comment:
what do you mean ?

Reply:
`IllegalArgumentException` is such a generic exception. I was just wondering if there is some way to be more specific. 

For example, let's suppose we got an `IllegalArgExp` with message 'Field not found`. Instead of raising a `BadRequestException` with the same message, we could do a `NotFoundException`.

That's what I meant by "specific case that could be captured and broken down into another less generic exception".

Reply:
The `IllegalArgumentException` is thrown in cases where the type is accepted but not the value, like expecting positive numbers and you give negative numbers. 
Replacing `IllegalArgumentException` with `BadRequestException` can be a good choice in certain contexts, particularly in web applications or APIs where the exception is meant to signal a client-side error.

Reply:
BTW , Other exceptions are already handled

Reply:
Good. My warning is just to avoid as much as possible returning generic messages to the customer that could have been more detailed.